src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs

```diff
@@ -38,5 +41,60 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIden
                 return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
             }
         }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
```

Comment:
BuildChallengeUrl now has additional functionality you should copy: https://github.com/aspnet/AspNetCore/blob/c89055b9b0664ca926a2ad7305969777ef32d319/src/Security/Authentication/OAuth/src/OAuthHandler.cs#L261-L276

Reply:
Done